pkg/apiserver/webhooks/datavolume-validate.go

```diff
@@ -199,7 +199,23 @@ func (wh *dataVolumeValidatingWebhook) validateDataVolumeSpec(request *v1beta1.A
 					return causes
 				}
 			}
-			err = controller.ValidateCanCloneSourceAndTargetSpec(&sourcePVC.Spec, spec.PVC)
+			sourceContentType, found := sourcePVC.Annotations[controller.AnnContentType]
+			if !found {
+				sourceContentType = string(cdiv1.DataVolumeKubeVirt)
+			}
+			targetContentType := spec.ContentType
+			if targetContentType == "" {
+				targetContentType = cdiv1.DataVolumeKubeVirt
+			}
+			if sourceContentType != string(targetContentType) {
+				causes = append(causes, metav1.StatusCause{
+					Type:    metav1.CauseTypeFieldValueInvalid,
+					Message: fmt.Sprintf("Source contentType (%s) and target contentType (%s) do not match", sourceContentType, targetContentType),
+					Field:   field.Child("PVC").String(),
+				})
+				return causes
+			}
+			err = controller.ValidateCanCloneSourceAndTargetSpec(&sourcePVC.Spec, spec.PVC, targetContentType)
```

Comment:
woow, this validateDataVolumeSpec might be a good candidate for some easy refactoring - dividing into small meaning methods, 
and this new addition would be something like validateMatchingContentType